utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -383,6 +420,244 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
     pub fn get_u32(&self, code_point: u32) -> u32 {
         self.get(code_point).into()
     }
+
+    /// Returns a [`CodePointMapRange`] struct representing a range of code
+    /// points associated with the same value.
+    pub fn get_range(&self, start: u32) -> Option<CodePointMapRange> {
+        if CODE_POINT_MAX < start {
```

Comment:
FYI  I am not looking at the details of this port, except that we discussed the types of some of the variables here, and the sentinel values. Otherwise I trust that it's a fairly straight port.

Reply:
Yes, it's a fairly straight port.  All of the changes are minor:

- change of the sentinel values & local variable types (i32 -> u32)
- return an owned range struct rather than mutating the input Range object in ICU4J
- handle all Rust `Option`s (returned by any sequenced index lookup) and `Result`s by converting the function's return type to an Option of a range and handling any error path in the body's logic by directly returning None
- converting `do { } while ()` to `loop { ... if () { break; } }`
- similar conversions for pre-/post-increment operators
- any boolean conditional clause rewrites according to the preferences of the Rust formatter

Reply:
I was too confident in that previous comment. TIL the hard way that `continue` won't automatically advance to my `if` statement at the bottom of the condition-less `loop` block.  I used `loop` because Rust doesn't have a `do-while` loop (I guess most new languages don't?).  Of course, for `loop` and `while`, a `continue` statement advances to the top of the block.